Source/Renamer/KerbalRenamer.cs

```diff
@@ -83,42 +93,81 @@ public void Awake()
             {
                 cultureDescriptor = data.GetValue("cultureDescriptor");
             }
+
             ConfigNode[] cultureclub = data.GetNodes("Culture");
             for (int i = 0; i < cultureclub.Length; i++)
             {
                 Culture c = new Culture(cultureclub[i]);
                 ctemp.Add(c);
             }
+
             cultures = ctemp.ToArray();
 
+            // Doesn't appear to be necessary anymore, doesn't hurt to initialize the profile.
+            LoadProfile("1951");
+
             GameEvents.onKerbalAddComplete.Add(new EventData<ProtoCrewMember>.OnEvent(OnKerbalAdded));
-            GameEvents.onGameStateCreated.Add(new EventData<Game>.OnEvent(OnGameCreated));
         }
 
-        void OnGameCreated(Game game)
+        public void OnDestroy()
+        {
+            GameEvents.onKerbalAddComplete.Remove(OnKerbalAdded);
+        }
+
+        /// <summary>
+        /// API to get a random name NOT associated with a Kerbal. 
+        /// </summary>
+        /// <remarks>Planned to be used by Headlines for non-crew individuals.</remarks>
+        /// <remarks>Useful for Headlines.</remarks>
+        /// <param name="gender"></param>
+        /// <param name="culture">to store the nationality</param>
+        /// <param name="name">to store the name</param>
+        public static void RandomName(ProtoCrewMember.Gender gender, ref string culture, ref string name)
+        {
+            Randomizer.GenerateRandomName(gender, ref culture, ref name, cultures);
+        }
+
+        /// <summary>
+        /// API to rename a crew member (or applicant) to another random name. This does NOT handles per-name indexing in RP1 or Headlines.
+        /// </summary>
+        /// <remarks>Can be used by other mods to replace a name to a more culture-appropriate one.</remarks>
+        /// <param name="crewMember"></param>
+        public static void Rename(ProtoCrewMember crewMember)
         {
+            Randomizer.RenameProtoCrewMember(crewMember, cultures);
         }
 
-        public void OnKerbalAdded(ProtoCrewMember kerbal)
+        private void OnKerbalAdded(ProtoCrewMember kerbal)
         {
-            LogUtils.Log("OnKerbalAdded called for " + kerbal.name);
+            LoadProfile(RenamerCustomParams.ProfileName);
+
+            LogUtils.Log("[RENAMER][BUILD] OnKerbalAdded called for " + kerbal.name +
+                         $" using profile {RenamerCustomParams.ProfileName}");
```

Comment:
Do not add extra tags to log calls since LogUtils already adds [KerbalRenamer] to them. The weird part is that `LogUtils.Log` never does anything but that's a separate issue.
Also it's better to use string interpolation on the entire string.

Reply:
I noticed that I had a mixture of `LogUtil.` and `KSPLog.`, Converts to `KSPLog` for consistency or the other way around?

Reply:
My IDE seems to suggest that KSPLog is underneath the `print()` calls in other RO codebases.

Reply:
TBH, I've never seen KSPLog used anywhere. Usually when the mod has it's own logging util, it's best to use that. I think the weirdness of LogUtil should be tackled in a separate PR.